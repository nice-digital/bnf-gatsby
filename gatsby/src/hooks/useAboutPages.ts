import { useStaticQuery, graphql } from "gatsby";

import { type MenuPageLink, type SlugAndTitle } from "@/utils";

export interface AboutPages {
	allBnfAboutSection: {
		aboutSectionPages: SlugAndTitle[];
	};
	allBnfCautionaryAndAdvisoryGuidance: {
		labelsGuidancePages: SlugAndTitle[];
	};
}

const slugToHref = ({ title, slug }: SlugAndTitle): MenuPageLink => ({
	title,
	href: `/about/${slug}/`,
});

/**
 * Gets the lists of pages for the about section by combining:
 * - about sections
 * - cautionary/advisory labels
 * - cautionary/advisory guidance page(s)
 * from the feed.
 *
 * @returns The consolidated list of about pages
 */
export const useAboutPages = (): MenuPageLink[] => {
	const { allBnfAboutSection, allBnfCautionaryAndAdvisoryGuidance } =
		useStaticQuery<AboutPages>(
			graphql`
				query AboutPages {
					allBnfAboutSection(sort: { order: ASC }) {
						aboutSectionPages: nodes {
							slug
							title
						}
					}
					allBnfCautionaryAndAdvisoryGuidance {
						labelsGuidancePages: nodes {
							title
							slug
						}
					}
				}
			`
		);

	return [
		...allBnfAboutSection.aboutSectionPages.map(slugToHref),
		{ title: "Labels", href: "/about/labels/" },
		...allBnfCautionaryAndAdvisoryGuidance.labelsGuidancePages.map(slugToHref),
	];
};
