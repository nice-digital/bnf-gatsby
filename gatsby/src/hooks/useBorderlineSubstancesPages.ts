import { useStaticQuery, graphql } from "gatsby";

import { SID, PHPID } from "@nice-digital/gatsby-source-bnf";

import { type MenuPageLink, type SlugAndTitle } from "@/utils";

export interface BorderlineSubstancesQueryResult {
	allBnfBorderlineSubstancesTaxonomyRoot: {
		pages: {
			slug: string;
			title: string;
		}[];
	};
}

const slugToHref = ({ title, slug }: SlugAndTitle): MenuPageLink => ({
	title,
	href: `/borderline-substances/${slug}/`,
});

/**
 * Gets the lists of pages for the borderline substances section
 *
 * @returns The consolidated list of borderline substances pages
 */
export const useBorderlineSubstancesPages = (): MenuPageLink[] => {
	const { allBnfBorderlineSubstancesTaxonomyRoot } =
		useStaticQuery<BorderlineSubstancesQueryResult>(
			graphql`
				{
					allBnfBorderlineSubstancesTaxonomyRoot {
						pages: nodes {
							slug
							title
						}
					}
				}
			`
		);

	return allBnfBorderlineSubstancesTaxonomyRoot.pages.map(slugToHref);
};
