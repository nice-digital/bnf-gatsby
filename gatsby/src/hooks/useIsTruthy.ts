import { useCallback } from "react";

// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
export const useIsTruthy = () =>
	useCallback(
		<TMaybe>(maybeT: TMaybe | null): maybeT is TMaybe => !!maybeT,
		[]
	);
