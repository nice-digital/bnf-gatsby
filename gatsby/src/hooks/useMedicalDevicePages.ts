import { useStaticQuery, graphql } from "gatsby";

import { type MenuPageLink, type SlugAndTitle } from "@/utils";

export interface MedicalDevicesQueryResult {
	allBnfMedicalDevice: {
		pages: SlugAndTitle[];
	};
}

const slugToHref = ({ title, slug }: SlugAndTitle): MenuPageLink => ({
	title,
	href: `/medical-devices/${slug}/`,
});

/**
 * Gets the lists of pages for the medical devices section
 *
 * @returns The consolidated list of medical devices pages
 */
export const useMedicalDevicePages = (): MenuPageLink[] => {
	const { allBnfMedicalDevice } = useStaticQuery<MedicalDevicesQueryResult>(
		graphql`
			{
				allBnfMedicalDevice(sort: { fields: title, order: ASC }) {
					pages: nodes {
						slug
						title
					}
				}
			}
		`
	);

	return allBnfMedicalDevice.pages.map(slugToHref);
};
