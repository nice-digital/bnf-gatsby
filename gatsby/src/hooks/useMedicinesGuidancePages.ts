import { useStaticQuery, graphql } from "gatsby";

import { type MenuPageLink, type SlugAndTitle } from "@/utils";

export interface MedicinesGuidanceQueryResult {
	allBnfGuidance: {
		pages: SlugAndTitle[];
	};
}

const slugToHref = ({ title, slug }: SlugAndTitle): MenuPageLink => ({
	title,
	href: `/medicines-guidance/${slug}/`,
});

/**
 * Gets the lists of pages for the medicines guidance section
 *
 * @returns The consolidated list of medicines guidance pages
 */
export const useMedicinesGuidancePages = (): MenuPageLink[] => {
	const { allBnfGuidance } = useStaticQuery<MedicinesGuidanceQueryResult>(
		graphql`
			{
				allBnfGuidance(sort: { fields: order, order: ASC }) {
					pages: nodes {
						slug
						title
					}
				}
			}
		`
	);

	return allBnfGuidance.pages.map(slugToHref);
};
