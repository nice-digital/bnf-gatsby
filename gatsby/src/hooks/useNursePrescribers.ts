import { useStaticQuery, graphql } from "gatsby";

import { SlugAndTitle, type MenuPageLink } from "@/utils";

export interface NursePrescribers {
	allBnfNursePrescribersFormularyIntroduction: {
		introduction: SlugAndTitle[];
	};
	allBnfNursePrescribersFormularyTreatmentSummary: {
		summaries: SlugAndTitle[];
	};
}

const slugToHref = ({ title, slug }: SlugAndTitle): MenuPageLink => ({
	title,
	href: `/nurse-prescribers-formulary/${slug}/`,
});

/**
 * Gets the lists of pages for the nurse prescribers formulary by combining:
 * - treatment summaries
 * - introduction
 * from the feed.
 *
 * @returns both the consolidated list for the details page menu, and two separate lists for the index page.
 */
export const useNursePrescribers = (): {
	menuList: MenuPageLink[];
	aboutList: MenuPageLink[];
	treatmentSummariesList: MenuPageLink[];
} => {
	const {
		allBnfNursePrescribersFormularyIntroduction,
		allBnfNursePrescribersFormularyTreatmentSummary,
	} = useStaticQuery<NursePrescribers>(
		graphql`
			query NursePrescribers {
				allBnfNursePrescribersFormularyIntroduction {
					introduction: nodes {
						slug
						title
					}
				}
				allBnfNursePrescribersFormularyTreatmentSummary {
					summaries: nodes {
						slug
						title
					}
				}
			}
		`
	);

	const menuList = [
		...allBnfNursePrescribersFormularyTreatmentSummary.summaries.map(
			slugToHref
		),
		...allBnfNursePrescribersFormularyIntroduction.introduction.map(slugToHref),
	];

	const aboutList = [
		...allBnfNursePrescribersFormularyTreatmentSummary.summaries
			.filter((summary) => summary.slug == "general-guidance")
			.map(slugToHref),
		...allBnfNursePrescribersFormularyIntroduction.introduction.map(slugToHref),
	];

	const treatmentSummariesList = [
		...allBnfNursePrescribersFormularyTreatmentSummary.summaries.map(
			slugToHref
		),
	];

	return { menuList, aboutList, treatmentSummariesList };
};
