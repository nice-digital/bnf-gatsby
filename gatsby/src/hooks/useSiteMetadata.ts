import { useStaticQuery, graphql } from "gatsby";

interface SiteMetaData {
	/** Whether the current site is BNF or BNFC */
	isBNF: boolean;
	/** Absolute URL to the site homepage, without trailing slash */
	siteUrl: string;
	/** Short site title e.g. "BNF" or "BNFC" */
	siteTitleShort: string;
	/** Full site title e.g. "British National Formulary" or "British National Formulary for Children" */
	siteTitleLong: string;
}

interface Site {
	site: {
		siteMetadata: SiteMetaData;
	};
}

/**
 * Gets the site metadata, as defined in gatsby-config
 *
 * @returns The site metadata
 */
export const useSiteMetadata = (): SiteMetaData => {
	const { site } = useStaticQuery<Site>(
		graphql`
			query SiteMetaData {
				site {
					siteMetadata {
						isBNF
						siteUrl
						siteTitleShort
						siteTitleLong
					}
				}
			}
		`
	);
	return site.siteMetadata;
};
