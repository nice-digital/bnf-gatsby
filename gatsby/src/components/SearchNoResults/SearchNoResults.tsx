import { Link } from "gatsby";
import { type FC } from "react";

import { useSiteMetadata } from "@/hooks/useSiteMetadata";

export interface SearchNoResultsProps {
	searchText: string;
}

export const SearchNoResults: FC<SearchNoResultsProps> = ({ searchText }) => {
	const { isBNF } = useSiteMetadata();

	return (
		<>
			<section aria-labelledby="browse-for-content">
				<h3 id="browse-for-content">Browse for content</h3>
				<p>Try browsing for content with our:</p>
				<ul aria-label="Ways to browse for content">
					<li>
						<Link to={"/drugs/"}>Drugs A to Z list</Link>
					</li>
					<li>
						<Link to={"/treatment-summaries/"}>
							Treatment summaries A to Z list
						</Link>
					</li>
					<li>
						<Link to={"/interactions/"}>Interactions A to Z list</Link>
					</li>
				</ul>
			</section>
			<section aria-labelledby="try-other-services">
				<h3 id="try-other-services">Check our other services</h3>
				<p>
					See if there are search results for <b>{searchText}</b> on our other
					services:
				</p>
				<ul aria-label={`Search for "${searchText}" on our other services`}>
					<li>
						<a
							rel="noreferrer"
							target="_blank"
							href={`https://bnf${
								isBNF ? "c" : ""
							}.nice.org.uk/search/?q=${searchText}&sp=on`}
						>
							{isBNF ? "BNFC" : "BNF"}
						</a>
					</li>
					<li>
						<a
							rel="noreferrer"
							target="_blank"
							href={`https://www.nice.org.uk/search?om=[{%22ndt%22:[%22Guidance%22]}]&ps=15&q=${searchText}&sp=on`}
						>
							NICE guidance
						</a>
					</li>
					<li>
						<a
							rel="noreferrer"
							target="_blank"
							href={`https://cks.nice.org.uk/search/?q=${searchText}&sp=on`}
						>
							CKS
						</a>
					</li>
				</ul>
				<section aria-labelledby="what-are-other-services">
					<h4 id="what-are-other-services">What are our other services?</h4>
					<h5>
						{isBNF
							? "British National Formulary for Children (BNFC)"
							: "British National Formulary (BNF)"}
					</h5>
					<p>
						Drug and prescribing information for healthcare professionals. A
						reference for correct dosage, indication, interaction and side
						effects of drugs.
					</p>
					<h5>NICE guidance</h5>
					<p>
						We use the best available evidence to develop recommendations that
						guide decisions in health, public health and social care.
					</p>
					<h5>Clinical Knowledge Summaries (CKS)</h5>
					<p>
						Concise, accessible summaries of current evidence for primary care
						professionals. Information is presented as patient scenarios or
						clinical presentations with answers and links to supporting
						evidence.
					</p>
				</section>
			</section>
		</>
	);
};
