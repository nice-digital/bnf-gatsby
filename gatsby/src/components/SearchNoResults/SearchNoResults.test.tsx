import { render } from "@testing-library/react";

import { useSiteMetadata } from "@/hooks/useSiteMetadata";

import { SearchNoResults } from "./SearchNoResults";

const useSiteMetadataMock = useSiteMetadata as jest.Mock;

describe("SearchNoResults", () => {
	it("should match snapshot for BNF site", () => {
		useSiteMetadataMock.mockImplementationOnce(() => ({ isBNF: true }));

		render(<SearchNoResults searchText="aspirin" />);

		expect(document.body).toMatchSnapshot();
	});

	it("should match snapshot for BNFC site", () => {
		useSiteMetadataMock.mockImplementationOnce(() => ({ isBNF: false }));

		render(<SearchNoResults searchText="aspirin" />);

		expect(document.body).toMatchSnapshot();
	});
});
