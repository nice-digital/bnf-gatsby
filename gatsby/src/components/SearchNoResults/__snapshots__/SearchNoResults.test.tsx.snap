// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchNoResults should match snapshot for BNF site 1`] = `
<body>
  <div>
    <section
      aria-labelledby="browse-for-content"
    >
      <h3
        id="browse-for-content"
      >
        Browse for content
      </h3>
      <p>
        Try browsing our:
      </p>
      <ul
        aria-label="Ways to browse for content"
      >
        <li>
          <a
            href="/drugs/"
          >
            drugs A to Z list
          </a>
        </li>
        <li>
          <a
            href="/treatment-summaries/"
          >
            treatment summaries A to Z list
          </a>
        </li>
        <li>
          <a
            href="/interactions/"
          >
            interactions A to Z list
          </a>
        </li>
      </ul>
    </section>
    <section
      aria-labelledby="try-other-services"
    >
      <h3
        id="try-other-services"
      >
        Check our other services
      </h3>
      <p>
        See if there are search results for 
        <b>
          aspirin
        </b>
         on our other services:
      </p>
      <ul
        aria-label="Search for \\"aspirin\\" on our other services"
      >
        <li>
          <a
            href="https://bnfc.nice.org.uk/search/?q=aspirin&sp=on"
            rel="noreferrer"
            target="_blank"
          >
            BNFC
          </a>
        </li>
        <li>
          <a
            href="https://www.nice.org.uk/search?om=[{%22ndt%22:[%22Guidance%22]}]&ps=15&q=aspirin&sp=on"
            rel="noreferrer"
            target="_blank"
          >
            NICE guidance
          </a>
        </li>
        <li>
          <a
            href="https://cks.nice.org.uk/search/?q=aspirin&sp=on"
            rel="noreferrer"
            target="_blank"
          >
            CKS
          </a>
        </li>
      </ul>
      <section
        aria-labelledby="what-are-other-services"
      >
        <h4
          id="what-are-other-services"
        >
          What are our other services?
        </h4>
        <h5>
          British National Formulary for Children (BNFC)
        </h5>
        <p>
          Information on the selection, prescribing, dispensing and administration of drugs to children.
        </p>
        <h5>
          NICE guidance
        </h5>
        <p>
          We use the best available evidence to develop recommendations that guide decisions in health, public health and social care.
        </p>
        <h5>
          Clinical Knowledge Summaries (CKS)
        </h5>
        <p>
          Concise, accessible summaries of current evidence for primary care professionals. Information is presented as patient scenarios or clinical presentations with answers and links to supporting evidence.
        </p>
      </section>
    </section>
  </div>
</body>
`;

exports[`SearchNoResults should match snapshot for BNFC site 1`] = `
<body>
  <div>
    <section
      aria-labelledby="browse-for-content"
    >
      <h3
        id="browse-for-content"
      >
        Browse for content
      </h3>
      <p>
        Try browsing our:
      </p>
      <ul
        aria-label="Ways to browse for content"
      >
        <li>
          <a
            href="/drugs/"
          >
            drugs A to Z list
          </a>
        </li>
        <li>
          <a
            href="/treatment-summaries/"
          >
            treatment summaries A to Z list
          </a>
        </li>
        <li>
          <a
            href="/interactions/"
          >
            interactions A to Z list
          </a>
        </li>
      </ul>
    </section>
    <section
      aria-labelledby="try-other-services"
    >
      <h3
        id="try-other-services"
      >
        Check our other services
      </h3>
      <p>
        See if there are search results for 
        <b>
          aspirin
        </b>
         on our other services:
      </p>
      <ul
        aria-label="Search for \\"aspirin\\" on our other services"
      >
        <li>
          <a
            href="https://bnf.nice.org.uk/search/?q=aspirin&sp=on"
            rel="noreferrer"
            target="_blank"
          >
            BNF
          </a>
        </li>
        <li>
          <a
            href="https://www.nice.org.uk/search?om=[{%22ndt%22:[%22Guidance%22]}]&ps=15&q=aspirin&sp=on"
            rel="noreferrer"
            target="_blank"
          >
            NICE guidance
          </a>
        </li>
        <li>
          <a
            href="https://cks.nice.org.uk/search/?q=aspirin&sp=on"
            rel="noreferrer"
            target="_blank"
          >
            CKS
          </a>
        </li>
      </ul>
      <section
        aria-labelledby="what-are-other-services"
      >
        <h4
          id="what-are-other-services"
        >
          What are our other services?
        </h4>
        <h5>
          British National Formulary (BNF)
        </h5>
        <p>
          Information on the selection, prescribing, dispensing and administration of drugs.
        </p>
        <h5>
          NICE guidance
        </h5>
        <p>
          We use the best available evidence to develop recommendations that guide decisions in health, public health and social care.
        </p>
        <h5>
          Clinical Knowledge Summaries (CKS)
        </h5>
        <p>
          Concise, accessible summaries of current evidence for primary care professionals. Information is presented as patient scenarios or clinical presentations with answers and links to supporting evidence.
        </p>
      </section>
    </section>
  </div>
</body>
`;
